init.lua

```diff
@@ -135,23 +133,54 @@ local function item_has_groups(item_groups, groups)
 	return true
 end
 
-local function groups_to_item(groups)
-	if #groups == 1 then
-		local group = groups[1]
-		if group_stereotypes[group] then
-			return group_stereotypes[group]
-		elseif reg_items["default:" .. group] then
-			return "default:" .. group
+local function get_group_items(str)
+	local group_string = str:sub(7)
+	if group_items[group_string] then
+		return group_items[group_string] --group_items[group][name] = def
+	end
+	local groups = group_string:split(",")
+	local list
+	for i = 1, #groups do
+		local group = groups[i]
+		if not group_items[group] then
+			list = false
+			break
+		elseif not list then
+			list = {}
+			for item, def in pairs(group_items[group]) do
+				list[item] = def
+			end
+		else
+			for item, def in pairs(list) do
+				if not group_items[group][item] then
+					list[item] = nil
+				end
+			end
 		end
 	end
+	if list and not next(list) then
+		list = false
+	end
+	group_items[group_string] = list
+	return list
+end
+craftguide.get_group_items = get_group_items
 
-	for name, def in pairs(reg_items) do
-		if item_has_groups(def.groups, groups) then
-			return name
-		end
+local function groups_to_item(str)
+	local group_string = str:sub(7)
```

Comment:
Use the name `group`.

Reply:
same as above. This group string is not resolved and can be 'flower,color_red'. `group` should be used for resolved groups only.